src/client/sandbox/storages/wrapper.js

```diff
@@ -19,219 +19,228 @@ function getWrapperMethods () {
-    this.on                = (ev, handler) => this.eventEmitter.on(ev, handler);
-    this.emit              = (ev, e) => this.eventEmitter.emit(ev, e);
-    this.nativeStorage     = nativeStorage;
-    this.nativeStorageKey  = nativeStorageKey;
-    this.lastState         = null;
-    this.window            = window;
-    this.initialProperties = [];
-    this.wrapperMethods    = [];
-    this.context           = window;
-
-    this.STORAGE_CHANGED_EVENT = 'hammerhead|event|storage-changed';
-    this.EMPTY_OLD_VALUE_ARG   = isIE ? '' : null;
-
-    const getAddedProperties = () => {
-        // NOTE: The standard doesn't regulate the order in which properties are enumerated.
-        // But we rely on the fact that they are enumerated in the order they were created in all the supported browsers.
-        // In this case we cannot use Object.getOwnPropertyNames
-        // because the enumeration order in Android 5.1 is different from all other browsers.
-        const properties = [];
-
-        for (const property in this) {
-            if (this.hasOwnProperty(property) && this.initialProperties.indexOf(property) === -1)
-                properties.push(property);
-        }
-
-        return properties;
-    };
-
-    nativeMethods.objectDefineProperty.call(window.Object, this, 'length', {
-        get: () => getAddedProperties().length,
-        set: () => void 0
-    });
-
-    const loadStorage = storage => {
-        if (!storage)
-            storage = this.nativeStorage[this.nativeStorageKey];
-
-        storage = JSON.parse(storage || '[[],[]]');
-
-        for (let i = 0; i < storage[KEY].length; i++)
-            this[storage[KEY][i]] = storage[VALUE][i];
-    };
-
-    const raiseStorageChanged = (key, oldValue, newValue) => {
-        let url = null;
-
-        try {
-            const parsedContextUrl = parseProxyUrl(this.context.location.toString());
-
-            url = parsedContextUrl ? parsedContextUrl.destUrl : destLocation.get();
-        }
-        catch (e) {
-            this.context = this.window;
-            url          = destLocation.get();
-        }
-
-        this.emit(this.STORAGE_CHANGED_EVENT, { key, oldValue, newValue, url });
-    };
-
-    const checkStorageChanged = () => {
-        const currentState = this.getCurrentState();
-
-        for (let i = 0; i < this.lastState[KEY].length; i++) {
-            const lastStateKey   = this.lastState[KEY][i];
-            const lastStateValue = this.lastState[VALUE][i];
-
-            const keyIndex = currentState[KEY].indexOf(lastStateKey);
-
-            if (keyIndex !== -1) {
-                if (currentState[VALUE][keyIndex] !== lastStateValue)
-                    raiseStorageChanged(currentState[KEY][keyIndex], lastStateValue, currentState[VALUE][keyIndex]);
-
-                currentState[KEY].splice(keyIndex, 1);
-                currentState[VALUE].splice(keyIndex, 1);
+export default class StorageWrapper {
```

Comment:
Why do we not extend this class from EventEmitter?

Reply:
We change prototype of `StorageWrapper`:
```js
StorageWrapper.prototype = Storage.prototype;
```

If `StorageWrapper` inherit from `EventEmitter` then we will get an `TypeError("Cannot call a class as a function")` from Babel's `_classCallCheck` common function.
Because `EventEmitter` was cleaned from prototype chain.